asubFunctionsApp/src/copyArgA.c

```diff
@@ -30,12 +30,12 @@ static long copyArgA(aSubRecord *prec)
          errlogPrintf("%s incorrect input type. ", prec->name);
 		 return -1;
 	}
+    
 	memset(str_out, '\0', prec->nova); /* pad with NULL */
-    for(n=0; n<prec->noa && n<prec->nova; ++n)
+    for(n=0; n<prec->noa && n<prec->nea; ++n)
```

Comment:
Copying up to nea will avoid the non-NULL termination issue, but you should still check prec->nova as if nea > nova you will get a memory overwrite